hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -292,8 +294,16 @@ protected void setCommitMetadata(HoodieWriteMetadata<HoodieData<WriteStatus>> re
   @Override
   protected void commit(HoodieWriteMetadata<HoodieData<WriteStatus>> result) {
     context.setJobStatus(this.getClass().getSimpleName(), "Commit write status collect: " + config.getTableName());
-    commit(result.getWriteStatuses(), result, result.getWriteStats().isPresent()
-        ? result.getWriteStats().get() : result.getWriteStatuses().map(WriteStatus::getStat).collectAsList());
+    if (!config.getBoolean(WRITE_IGNORE_FAILED) && result.getWriteStatuses().filter(status -> status.getErrors().size() > 0).count() > 0) {
```

Comment:
How about move those fix logical into `BaseCommitActionExecutor#commit(HoodieData<WriteStatus> writeStatuses, HoodieWriteMetadata<O> result, List<HoodieWriteStat> writeStats)`?

Reply:
OK，move it latter~